src/loader.c

```diff
@@ -157,8 +159,30 @@ static void *imv_loader_bg_new_img(void *data)
   pthread_mutex_lock(&ldr->lock);
   char *path = strdup(ldr->path);
   pthread_mutex_unlock(&ldr->lock);
+  FREE_IMAGE_FORMAT fmt = NULL;
+  FIBITMAP *bmp = NULL;
+
+  int load_from_memory = 0;
+  if(ldr->stdin_buffer != NULL && strcmp(path, "stdin") == 0){
+    load_from_memory = 1;
+  }
 
-  FREE_IMAGE_FORMAT fmt = FreeImage_GetFileType(path, 0);
+  if(load_from_memory == 1){
```

Comment:
At this point for both the path and the stdin case we should just be determining the filetype and setting fmt appropriately.